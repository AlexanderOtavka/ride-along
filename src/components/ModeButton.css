/**
 * ModeButton.css
 *
 * Created by Zander Otavka on 6/3/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

.ModeButton {
  position: relative;
  display: inline-block;
  width: 56px;
  height: 48px;
}

.ModeButton__input {
  position: absolute;
  visibility: hidden;
}

.ModeButton__button {
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  display: flex;
  justify-content: center;
  align-items: center;
  background-color: #F2F2F2;
}

.ModeButton__input:checked ~ .ModeButton__button {
  background-color: rgba(242, 242, 242, 0.3);
}

.ModeButton__button > svg {
  width: 28px;
  height: 28px;
}

.ModeButton__button--request > svg {
  fill: #F2C94C;
}

.ModeButton__button--offer > svg {
  fill: #219653;
}
.ModeButton__input:checked ~ .ModeButton__button > svg {
  fill: #F2F2F2;
}
