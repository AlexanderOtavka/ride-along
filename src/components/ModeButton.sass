/**
 * ModeButton.sass
 *
 * Created by Zander Otavka on 6/3/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

@import "style/colors"
@import "style/layout"
@import "style/effects"

.modeButton
  position: relative
  display: inline-block
  width: 56px
  height: 48px

  &.request .ripple
    background-color: $primary-yellow

  &.offer .ripple
    background-color: $primary-green

.input
  position: absolute
  visibility: hidden

.button
  position: absolute
  top: 0
  right: 0
  bottom: 0
  left: 0
  display: flex
  justify-content: center
  align-items: center
  box-shadow: none

  .input:checked ~ &
    box-shadow: $shadow1
    pointer-events: none

  &::after
    +fullbleed

    content: ""
    z-index: -1
    background-color: $off-white
    opacity: 0.3
    transition: opacity 400ms
    will-change: opacity

    .input:checked ~ &
      opacity: 1

.icon
  width: 28px
  height: 28px
  fill: $off-white

  .input:checked ~ .button &
    .modeButton.request &
      fill: $primary-yellow

    .modeButton.offer &
      fill: $primary-green
