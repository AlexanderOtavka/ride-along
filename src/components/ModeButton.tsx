/**
 * @file ModeButton.tsx
 *
 * Created by Zander Otavka on 6/3/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import * as React from "react"

import bemHelper from "../util/bemHelper"
import "./ModeButton.css"

interface ModeButtonProps {
  name: string
  mode: string
  defaultChecked?: boolean
  imageSrc: string
}

const bem = bemHelper("ModeButton")

export default function ModeButton({
  mode,
  defaultChecked = false,
  ...props,
}: ModeButtonProps) {
  return (
    <label className={bem("")}>
      <input
        className={bem("input")}
        type="radio"
        name={props.name}
        value={mode}
        defaultChecked={defaultChecked}
      />

      <div className={bem("button", mode)}>
        <img className={bem("image", mode)} src={props.imageSrc} />
      </div>
    </label>
  )
}
