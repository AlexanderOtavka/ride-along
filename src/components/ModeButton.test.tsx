/**
 * @file ModeButton.test.tsx
 *
 * Created by Zander Otavka on 6/6/17.
 * Copyright (C) 2016  Grinnell AppDev.
 *
 * @license
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

import React from "react"
import renderer from "react-test-renderer"

import ModeButton from "./ModeButton"

it("without a selectedMode", () => {
  const component = renderer.create(
    <ModeButton name="foo" mode="request" image={<svg />} />
  )

  expect(component.toJSON()).toMatchSnapshot()
})

it("when selectedMode is different from mode", () => {
  const component = renderer.create(
    <ModeButton
      name="foo"
      mode="offer"
      selectedMode="request"
      image={<svg />}
    />
  )

  expect(component.toJSON()).toMatchSnapshot()
})

it("when selectedMode matches mode", () => {
  const component = renderer.create(
    <ModeButton name="foo" mode="offer" selectedMode="offer" image={<svg />} />
  )

  expect(component.toJSON()).toMatchSnapshot()
})
