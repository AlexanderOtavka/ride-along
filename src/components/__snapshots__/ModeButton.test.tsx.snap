// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is not selected when selectedMode is different from mode 1`] = `
<label
  className="modeButton"
>
  <input
    className="input"
    defaultChecked={false}
    name="foo"
    type="radio"
    value="offer"
  />
  <div
    className="button offer"
  >
    <svg />
  </div>
</label>
`;

exports[`is not selected without a selectedMode 1`] = `
<label
  className="modeButton"
>
  <input
    className="input"
    defaultChecked={false}
    name="foo"
    type="radio"
    value="request"
  />
  <div
    className="button request"
  >
    <svg />
  </div>
</label>
`;

exports[`is selected when selectedMode matches mode 1`] = `
<label
  className="modeButton"
>
  <input
    className="input"
    defaultChecked={true}
    name="foo"
    type="radio"
    value="offer"
  />
  <div
    className="button offer"
  >
    <svg />
  </div>
</label>
`;
